src/ol_openedx_canvas_integration/views.py

```diff
@@ -101,8 +104,9 @@ def list_canvas_assignments(request, course_id):  # noqa: ARG001
     """List Canvas assignments"""
     course_key = CourseLocator.from_string(course_id)
     course = get_course_by_id(course_key)
-    client = CanvasClient(canvas_course_id=course.canvas_course_id)
-    if not course.canvas_course_id:
+    canvas_course_id = get_canvas_course_id(course)
+    client = CanvasClient(canvas_course_id=canvas_course_id)
```

Comment:
Yeah, I checked that. I just mentioned it because of the validation. We can either initialize it after the validation or remove the validation if it is not needed.